### ISSUE TITLE
Validity not updated for some of the controls in a repeat 

### ISSUE THREAD

**Issue by avernet** [orbeon/orbeon-forms#5369]
Publish [this form](https://github.com/orbeon/orbeon-forms/files/9006210/form.txt), test, click "+", type "b", enter ⇒ the first 2 inputs are marked invalid, but the third isn't, while it should. (Remove the "b", type "abc", press enter, type "c", press enter ⇒ still, only the first 2 inputs are marked invalid.)
 
[+1 from customer](https://3.basecamp.com/3600924/buckets/16915667/messages/4316281306)

**Comment by ebruchez** [comment]
The constraint is:

```xpath
count(
    distinct-values(
        for $id in fr:control-string-value('id', false())
        return substring($id, 1, 3)
    )
) < 2
```

Now `fr:control-string-value()` always only returns the value of a single control. Further, it is an constraint in the *model*, and here looks for a value in the *view*, which is a dangerous dependency.

At best, this wouldn't work for the reason that it always return only a single item.

Trying `$id` searches, and finds, as expected, the closest control. This works as expected but doesn't satisfy the requirement of finding all the values.

Trying the `fr:control-typed-value()' function, which is able to return more than one item (confusingly), I do obtain all the values. However, the problem of the model/view dependency is still present.

**Comment by ebruchez** [comment]
The following works:

```
count(
    distinct-values(
        let $a := fr:control-typed-value('id', false())
        return
            for $i in (1 to array:size($a))
            return substring(array:get($a, $i), 1, 3)
    )
) < 2
```

**Comment by ebruchez** [comment]
It would be good, however, to have a way to express accessing values, for validation, that are in the model. We can do this with `$foo`, but, as noted, this resolves to "closest", and without options. In many cases, this is what we want, but in others, like here, not.

**Comment by ebruchez** [comment]
In fact, we noticed that `fr:control-string-value()` returns all values ilke `fr:control-typed-value()`! That was not the original intent, and the doc is therefore incorrect vs. the code. But since this has been the case since that function was implemented with Orbeon Forms 2018.1, we can keep that behavior.

- [ ] update function signature for clarity
- [ ] update doc

**Comment by ebruchez** [comment]
Closing the bug as the original behavior was due to controls in the other repeat iterations not being visited.
